Applications/Spire/Include/Spire/Ui/AssociativeValueModel.hpp

```diff
@@ -0,0 +1,334 @@
+#ifndef SPIRE_ASSOCIATIVE_VALUE_MODEL_HPP
+#define SPIRE_ASSOCIATIVE_VALUE_MODEL_HPP
+#include <boost/signals2/shared_connection_block.hpp>
+#include "Spire/Ui/Ui.hpp"
+#include "Spire/Ui/LocalValueModel.hpp"
+
+namespace Spire {
+namespace Details {
+  struct AssociatedBooleanModel {
+    std::shared_ptr<BooleanModel> m_model;
+    boost::signals2::scoped_connection m_connection;
+  };
+}
+
+  /** 
+  * Associates BooleanModels with corresponding values, and ensures that
+  * only a single model has a value of true at one time.
+  */
+  template <typename T>
+  class AssociativeValueModel : public ValueModel<T> {
+    public:
+
+      /** Constructs an AssociativeValueModel. */
+      AssociativeValueModel();
+
+      /**
+      * Associates a BooleanModel iff it's not already associated.
+      * If there are no existing models the BooleanModel is set to true,
+      * otherwise, it's set to false.
+      * @param model The model to associate.
+      * @param value The associated value.
+      */
+      void associate(const std::shared_ptr<BooleanModel>& model,
+        const T& value);
```

Comment:
Don't reference arguments within a template, instead capture the argument in a ```using``` declaration and use that. In this case ```ValueModel<T>``` captures ```T``` via ```Type```.